### PR TITLE
Mccalluc/beanstalk docker

### PR DESCRIPTION
**Pull request by mccalluc**
This lets a user with AWS credentials in place spin up an Elastic Beanstalk instance. This path uses almost the same Dockerfile as the Refinery... except that Refinery mounts a volume at `/data`, while for Beanstalk the input data is just baked into the image that gets deployed. If file sizes get too large at some point, I think the solution would be keep the data on s3 or some other URL.

Besides just reviewing the changes, if you could checkout this branch and see if [the instructions](https://github.com/refinery-platform/heatmap-scatter-dash/blob/mccalluc/beanstalk-docker/README.md) make sense, and actually work, that would be cool. Thanks! 

(Down the road, the idea is that Refinery would also be able to launch (grow?) Beanstalks. Right now I'm thinking there will be something new, but analogous to `django_docker_engine`... This new thing is still using Docker, but that's really an internal. Beanstalk handles a range of bundles, not just Docker, and the new machinery would be about launching EB from Refinery.)

(Ignore the Pipfile -> requirements.txt change: On an earlier PR I had moved to Pipfile, and I think it's good, but things will be a little smoother when it's more mature.)

----------------

erg: Sorry, this isn't right: I've confused myself around the input.json. I think what's it's trying to do right now is "download" a file from a file:// url, and promptly resave it locally... Let's skip all of that, and instead, when deploying to aws, the arguments are not files but urls. (And maybe the script can push local files to s3 if need be?) We still need to make the distinction about whether /data is mounted or not, but only input.json will be baked in, not everything.

Code review at this point is still welcome: If you think I'm going entirely the wrong direction, but better to know sooner than later.

### CODE REVIEW

**Comment by mccalluc** [comment]
erg: Sorry, this isn't right: I've confused myself around the `input.json`. I think what's it's trying to do right now is "download" a file from a `file://` url, and promptly resave it locally... Let's skip all of that, and instead, when deploying to aws, the arguments are not files but urls. (And maybe the script can push local files to s3 if need be?) We still need to make the distinction about whether `/data` is mounted or not, but only `input.json` will be baked in, not everything.

Code review at this point is still welcome: If you think I'm going entirely the wrong direction, but to know sooner than later.

**Comment by mccalluc** [comment]
I think it's working now! Feedback very welcome. Next question is how to integrate this with Refinery.

File: context/Dockerfile.refinery [hunk 1]
```diff
@@ -0,0 +1 @@
+FROM mccalluc/heatmap_scatter_dash:v0.0.15
```
Comment (scottx611x) [inline context/Dockerfile.refinery]: Needs `VOLUME`
& 
use :latest ? or remove `:`

File: context/app_runner_aws.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    parser.add_argument(
+        '--diffs', nargs='+', metavar='CSV',
+        type=argparse.FileType('r'), default=[],
+        help='Read CSV or TSV files containing differential expression data.')
+    # During development, it's useful to be able to specify a high port.
+    return parser
+
+
+def input(file_urls=[], diff_urls=[]):
```
Comment (scottx611x) [inline context/app_runner_aws.py]: mutable default args?

File: context/app_runner_refinery.py [hunk 3]
```diff
@@ -60,8 +61,10 @@ def _download_files(self, urls, data_dir):
 
         for url in urls:
             try:
+                session = requests.Session()
+                session.mount('file://', FileAdapter())
```
Comment (scottx611x) [inline context/app_runner_refinery.py]: Might need to delete this?

**Pull request merged**